extension_helpers/_setup_helpers.py

```diff
@@ -7,18 +7,33 @@
 import os
 import sys
 import shutil
+import logging
 import subprocess
-from distutils import log
 from collections import defaultdict
-from distutils.core import Extension
 
-from setuptools import find_packages
-from setuptools.config import read_configuration
+from setuptools import Extension, find_packages
+from setuptools.command.build_ext import new_compiler
 
-from ._distutils_helpers import get_compiler
 from ._utils import import_file, walk_skip_hidden
 
-__all__ = ['get_extensions', 'pkg_config']
+__all__ = ['get_compiler', 'get_extensions', 'pkg_config']
+
+log = logging.getLogger(__name__)
+
+
+def get_compiler():
+    """
+    Determines the compiler that will be used to build extension modules.
+
+    Returns
+    -------
+    compiler : str
+        The compiler option specified for the build, build_ext, or build_clib
+        command; or the default compiler for the platform if none was
+        specified.
+
+    """
+    return new_compiler().compiler_type
 
```

Comment:
If this is the upshot of this change then +:100: because it will make #33 a lot easier.

Reply:
I mean at some point we can even deprecate ``get_compiler`` but no rush for that

Reply:
We would need to provide a way to switch on compiler / platform though as different compilers can use different flags etc.